examples/multisite_neuroimg_adapt/main.py

```diff
@@ -0,0 +1,129 @@
+"""
+Autism Detection: Domain Adaptation for Multi-Site Neuroimaging Data Analysis
+
+Reference:
+[1] Cameron Craddock, Yassine Benhajali, Carlton Chu, Francois Chouinard, Alan Evans, András Jakab, Budhachandra Singh Khundrakpam, John David Lewis, Qingyang Li, Michael Milham, Chaogan Yan, Pierre Bellec (2013). The Neuro Bureau Preprocessing Initiative: open sharing of preprocessed neuroimaging data and derivatives. In Neuroinformatics 2013, Stockholm, Sweden.
+
+[2] Abraham, A., Pedregosa, F., Eickenberg, M., Gervais, P., Mueller, A., Kossaifi, J., ... & Varoquaux, G. (2014). Machine learning for neuroimaging with scikit-learn. Frontiers in neuroinformatics, 14.
+
+[3] Zhou, S., Li, W., Cox, C.R., & Lu, H. (2020). Side Information Dependence as a Regularizer for Analyzing Human Brain Conditions across Cognitive Experiments. in AAAI 2020, New York, USA. https://ojs.aaai.org//index.php/AAAI/article/view/6179
+
+[4] Zhou, S. (2022). Interpretable Domain-Aware Learning for Neuroimage Classification (Doctoral dissertation, University of Sheffield). https://etheses.whiterose.ac.uk/31044/1/PhD_thesis_ShuoZhou_170272834.pdf
+"""
+import os
+import warnings
+
+import numpy as np
+import pandas as pd
+from config import get_cfg_defaults
+from nilearn.connectome import ConnectivityMeasure
+from nilearn.datasets import fetch_abide_pcp
+from sklearn.linear_model import RidgeClassifier
+from sklearn.metrics import accuracy_score
+from sklearn.preprocessing import OneHotEncoder
+
+from kale.interpret import visualize
+from kale.pipeline.multi_domain_adapter import CoIRLS
+from kale.utils.download import download_file_by_url
+
+
+def cross_validation(x, y, covariates, estimator, domain_adaptation=False):
```

Comment:
@shuo-zhou Sure this is not reusable?  To discuss this before merging.

Reply:
Other pipelines with dataloader and lightning trainer do not need this

Reply:
The issue is not whether some other APIs do not need this. It is about what some APIs/examples may need. A library's value is about what it can do, not what it cannot do.